storage/remote/client.go

```diff
@@ -267,28 +267,38 @@ func (c *Client) Store(ctx context.Context, req []byte, attempt int) error {
 	if err != nil {
 		// Errors from Client.Do are from (for example) network errors, so are
 		// recoverable.
-		return RecoverableError{err, defaultBackoff}
+		return WriteResponseStats{}, RecoverableError{err, defaultBackoff}
 	}
 	defer func() {
 		io.Copy(io.Discard, httpResp.Body)
 		httpResp.Body.Close()
 	}()
 
+	rs, err := ParseWriteResponseStats(httpResp)
+	if err != nil {
+		// There were some response header values we can't parse, so fail fast here.
+		return WriteResponseStats{}, err
+	}
+
 	//nolint:usestdlibvars
-	if httpResp.StatusCode/100 != 2 {
-		scanner := bufio.NewScanner(io.LimitReader(httpResp.Body, maxErrMsgLen))
-		line := ""
-		if scanner.Scan() {
-			line = scanner.Text()
-		}
-		err = fmt.Errorf("server returned HTTP status %s: %s", httpResp.Status, line)
+	if httpResp.StatusCode/100 == 2 {
+		return rs, nil
+	}
+
+	// Handling errors e.g. read potential error in the body.
+	scanner := bufio.NewScanner(io.LimitReader(httpResp.Body, maxErrMsgLen))
+	line := ""
+	if scanner.Scan() {
+		line = scanner.Text()
```

Comment:
question: Could this not get the full error message? Should we loop and add append more text, or maybe put an indication that we didn't get the whole thing (for example with ellipsis)?

Reply:
good point, will do